### PR TITLE
feat: Added StoreCollectionClient for listing Actors in Apify store

### PR DESCRIPTION
**Pull request by drobnikj**
- Added new methods for https://docs.apify.com/api/v2/#/reference/store/store-actors-collection
- Updated actions/checkout across GH workflows because https://github.com/actions/checkout/issues/1448#issuecomment-1705344036

### CODE REVIEW

File: src/apify_client/clients/resource_clients/store_collection.py [hunk 1]
```diff
@@ -0,0 +1,100 @@
...
+    ) -> ListPage[Dict]:
+        """List Actors in Apify store.
+
+        https://docs.apify.com/api/v2/#/reference/store/store-actors-collection/get-list-of-actors-in-store
+
+        Args:
+            limit (int, optional): How many Actors to list
+            offset (int, optional): What Actor to include as first when retrieving the list
+            search (str, optional): String to search by. The search runs on the following fields: title, name, description, username, readme.
+            sort_be (str, optional): Specifies the field by which to sort the results.
```
Comment (fnesveda) [inline src/apify_client/clients/resource_clients/store_collection.py:37]:
```suggestion
            sort_by (str, optional): Specifies the field by which to sort the results.
```

File: src/apify_client/clients/resource_clients/store_collection.py [hunk 2]
```diff
@@ -0,0 +1,100 @@
...
```
Comment (fnesveda) [inline src/apify_client/clients/resource_clients/store_collection.py:84]:
```suggestion
            sort_by (str, optional): Specifies the field by which to sort the results.
```

File: docs/docs.md [hunk 3]
```diff
@@ -6189,3 +6211,83 @@ List all webhook dispatches of a user.
...
+
+* **Parameters**
+
+  * **limit** (`int`, *optional*) – How many Actors to list
+
+  * **offset** (`int`, *optional*) – What Actor to include as first when retrieving the list
+
+  * **search** (`str`, *optional*) – String to search by. The search runs on the following fields: title, name, description, username, readme.
+
+  * **sort_be** (`str`, *optional*) – Specifies the field by which to sort the results.
```
Comment (fnesveda) [inline docs/docs.md:6239]:
```suggestion
  * **sort_by** (`str`, *optional*) – Specifies the field by which to sort the results.
```

File: docs/docs.md [hunk 4]
```diff
@@ -6189,3 +6211,83 @@ List all webhook dispatches of a user.
...
```
Comment (fnesveda) [inline docs/docs.md:6279]:
```suggestion
  * **sort_by** (`str`, *optional*) – Specifies the field by which to sort the results.
```

**Review by fnesveda — CHANGES_REQUESTED** [review]
Cool, just one typo in the docstrings

**Review by fnesveda — APPROVED** [review]

**Review by vdusek — APPROVED** [review]
LGTM

**Pull request merged**